spec/locales-currencies-tz.html

```diff
@@ -219,15 +254,34 @@ <h1>AvailableNamedTimeZoneIdentifiers ( ): a List of Time Zone Identifier Record
-          1. If _identifier_ is a Link name and _identifier_ is not *"UTC"*, then
-            1. Set _primary_ to the Zone name that _identifier_ resolves to, according to the rules for resolving Link names in the IANA Time Zone Database.
-            1. NOTE: An implementation may need to resolve _identifier_ iteratively.
+          1. NOTE: The algorithm steps below are intended to correspond to the behaviour of <code>icu::TimeZone::getIanaID()</code> in the International Components for Unicode (<a href="https://icu.unicode.org/">ICU</a>) and the processes for maintaining time zone identifier data in the Unicode Common Locale Data Repository (<a href="https://cldr.unicode.org">CLDR</a>).
+          1. If _identifier_ is a Link name in the IANA Time Zone Database and _identifier_ is not present in the "TZ" column of <code>zone.tab</code> of the IANA Time Zone Database, then
+            1. Let _zone_ be the Zone name that _identifier_ resolves to, according to the rules for resolving Link names in the IANA Time Zone Database.
+            1. If _zone_ starts with *"Etc/"*, then
+              1. Set _primary_ to _zone_.
+            1. Else,
+              1. Let _identifierCountryCode_ be the <a href="https://www.iso.org/glossary-for-iso-3166.html">ISO 3166-1 Alpha-2</a> country code whose territory contains the geographical area corresponding to _identifier_.
+              1. Let _zoneCountryCode_ be the ISO 3166-1 Alpha-2 country code whose territory contains the geographical area corresponding to _zone_.
+              1. If _identifierCountryCode_ is _zoneCountryCode_, then
+                1. Set _primary_ to _zone_.
+              1. Else,
+                1. Let _countryCodeLineCount_ be the number of lines in file <code>zone.tab</code> of the IANA Time Zone Database where the "country-code" column is _identifierCountryCode_.
+                1. If _countryCodeLineCount_ is 1, then
+                  1. Let _countryCodeLine_ be the line in file <code>zone.tab</code> of the IANA Time Zone Database where the "country-code" column is _identifierCountryCode_.
+                  1. Set _primary_ to the contents of the "TZ" column of _countryCodeLine_.
+                1. Else,
+                  1. Let _backzoneLinkLines_ be the subset of lines in the file <code>backzone</code> of the IANA Time Zone Database that start with either *"Link"* or *"#PACKRATLIST zone.tab Link"*.
+                  1. Assert: Exactly one line in _backzoneLinkLines_ contains _identifier_ in the second, space-delimited column after *"Link"* or *"#PACKRATLIST zone.tab Link"*.
+                  1. Let _backzoneLinkLine_ be the line in _backzoneLinkLines_ that contains _identifier_ in the second, space-delimited column after *"Link"* or *"#PACKRATLIST zone.tab Link"*.
+                  1. Set _primary_ to the contents of the first, space-delimited column in _backzoneLinkLine_ after *"Link"* or *"#PACKRATLIST zone.tab Link"*.
```

Comment:
I've wanted a StringSplit operation for a while ([IsWellFormedUnitIdentifier](https://tc39.es/ecma402/#sec-iswellformedunitidentifier) is practically begging for it), but in the absence of such I'd like to be crystal-clear in this kind of algorithm.
```suggestion
                  1. Let _backzone_ be *undefined*.
                  1. Let _backzoneLinkLines_ be the List of lines in the file <code>backzone</code> of the IANA Time Zone Database that start with either *"Link "* or *"#PACKRATLIST zone.tab Link "*.
                  1. For each element _line_ of _backzoneLinkLines_, do
                    1. If _line_ starts with *"#PACKRATLIST zone.tab "*, set _line_ to the substring of _line_ from 22.
                    1. Assert: _line_ starts with *"Link "*.
                    1. Let _i_ be 5.
                    1. Let _j_ be StringIndexOf(_line_, *" "*, _i_).
                    1. Assert: _j_ is not ~not-found~ and _j_ > _i_.
                    1. Let _k_ be StringIndexOf(_line_, *" "*, _j_ + 1).
                    1. If _k_ is ~not-found~, set _k_ to the length of _line_.
                    1. Assert: _k_ > _j_ + 1.
                    1. Let _alias_ be the substring of _line_ from _j_ + 1 to _k_.
                    1. If _alias_ is _identifier_, then
                      1. Assert: _backzone_ is *undefined*.
                      1. Set _backzone_ to the substring of _line_ from _i_ to _j_.
                  1. Assert: _backzone_ is not *undefined*.
                  1. Set _primary_ to _backzone_.
```
A worked example of e.g. Pacific/Truk against current tzdata in an editorial note would also be helpful.

Reply:
Oh wow that's hard to read. I wonder if it'd be better to introduce a string split AO into 402 in this PR and use it in this PR, then it could be re-used elsewhere in 402 later?

Reply:
Take a look at the commit I just pushed that includes a minimal string-split AO, borrowed from the lower half of the spec text of [`String.prototype.split`](https://tc39.es/ecma262/#sec-string.prototype.split). 

> A worked example of e.g. Pacific/Truk against current tzdata in an editorial note would also be helpful.

When you say "worked example" what do you mean?

Reply:
> Take a look at the commit I just pushed that includes a minimal string-split AO, borrowed from the lower half of the spec text of [`String.prototype.split`](https://tc39.es/ecma262/#sec-string.prototype.split).

And while we're at it, I refactored `IsWellFormedUnitIdentifier` to use the new AO.

Reply:
> When you say "worked example" what do you mean?

A note explaining something like this:
> This algorithm attempts to resolve Links to primary time zone identifiers without crossing the boundaries of <a href="https://www.iso.org/glossary-for-iso-3166.html">ISO 3166-1 Alpha-2</a> country codes, using data from files <code>zone.tab</code> and <code>backzone</code> of the IANA Time Zone Database as necessary.
> For example, if *"Pacific/Truk"* (in country code *"FM"*) is a Link to *"Pacific/Port_Moresby"* (in country code *"PG"*) with default configuration, then <code>zone.tab</code> will be checked for lines corresponding with country code *"FM"*. If there is only one such line, then the Zone of that line will be treated as the primary time zone identifier associated with *"Pacific/Truk"*. But otherwise, that data will be taken from the unique Link in <code>backzone</code> with source name *"Pacific/Truk"* (e.g., line *"Link Pacific/Chuuk Pacific/Truk"* will result in primary time zone identifier *"Pacific/Chuuk"*).

(or substitute any other example that requires use of `backzone`)

Reply:
OK, I added an editorial note in the latest commit. Let me know what you think.